### PR TITLE
feat(plugin-commands-patching): apply existed patch file when re-patch

### LINKED ISSUE #5632: When patch is executed for the second time, the previous patch is not applied in the workspace.
### pnpm version:
7.16.0
### Code to reproduce the issue:
I want to change the code in node_modules. I already had a patch a week ago, and it was successfully applied in node_modules. When I executed 'pnpm patch react-awesome-query-builder' for the second time, Terminal display `You can now edit the following folder: C:\Users\Real\AppData\Local\Temp\7f4be305f0bbc18c6825c91ffc1d4b7d`, When I entered this temporary workspace, I found that the previous patch had not been applied.
### Expected behavior:
The previous patch in the temporary workspace will be applied
### Actual behavior:
The previous patch in the temporary workspace not be applied
### Additional information:
my package.json: 
```
"pnpm": {
    "patchedDependencies": {
      "react-awesome-query-builder@5.3.1": "patches/react-awesome-query-builder@5.3.1.patch"
    }
  }
```
 - `node -v` prints: v16.13.0
 - Windows, macOS, or Linux?: Windows

### PR DESCRIPTION
**Pull request by await-ovo**
fix #5632 

* reuse existed patch file by default.
* add `--ignore-patches` option to ignore current patch files

### CODE REVIEW

File: packages/plugin-commands-patching/src/patch.ts [hunk 1]
```diff
@@ -35,16 +35,21 @@ export function help () {
           description: 'The package that needs to be modified will be extracted to this directory',
           name: '--edit-dir',
         },
+        {
+          description: 'Ignore existing patch files when patching',
+          name: '--ignore-patches',
```
Comment (zkochan) [inline packages/plugin-commands-patching/src/patch.ts]: There may be only one patch, so it should be singular. I suggest to use `--ignore-existing`

File: packages/plugin-commands-patching/src/writePackage.ts [hunk 2]
```diff
@@ -26,4 +34,17 @@ export async function writePackage (pkg: string, dest: string, opts: WritePackag
     filesResponse,
     force: true,
   })
+
+  if (!opts.isCommit && !opts.ignorePatches) {
+    const { rootProjectManifest } = opts
+    const existedPatchFile = dep.alias && dep.pref && rootProjectManifest?.pnpm?.patchedDependencies?.[`${dep.alias}@${dep.pref}`]
```
Comment (zkochan) [inline packages/plugin-commands-patching/src/writePackage.ts]: I think it should be `existingPatchFile`

File: packages/plugin-commands-patching/src/patch.ts (hunk 1 above)
Comment (await-ovo) [reply packages/plugin-commands-patching/src/patch.ts]: Thanks for your suggestion. I'll update it.

File: packages/plugin-commands-patching/src/writePackage.ts [hunk 3]
```diff
@@ -26,4 +34,17 @@ export async function writePackage (pkg: string, dest: string, opts: WritePackag
     filesResponse,
     force: true,
   })
+
+  if (!opts.isCommit && !opts.ignorePatches) {
```
Comment (zkochan) [inline packages/plugin-commands-patching/src/writePackage.ts]: In this case you won't need to alias and pref below
```suggestion
  if (!opts.isCommit && !opts.ignorePatches && dep.alias && dep.pref) {
```

File: packages/plugin-commands-patching/src/writePackage.ts [hunk 4]
```diff
@@ -26,4 +34,17 @@ export async function writePackage (pkg: string, dest: string, opts: WritePackag
...
+    const { rootProjectManifest } = opts
```
Comment (zkochan) [inline packages/plugin-commands-patching/src/writePackage.ts]: you only need to use rootProjectManifest once. No need to declare a new variable for it.

File: packages/plugin-commands-patching/src/writePackage.ts (hunk 2 above)
Comment (zkochan) [reply packages/plugin-commands-patching/src/writePackage.ts]:
```suggestion
    const existingPatchFile = opts.rootProjectManifest?.pnpm?.patchedDependencies?.[`${dep.alias}@${dep.pref}`]
```

File: packages/plugin-commands-patching/src/writePackage.ts (hunk 4 above)
Comment (await-ovo) [reply packages/plugin-commands-patching/src/writePackage.ts]: already updated now and added changeset :)

**Review by zkochan — APPROVED** [review]

**Comment by await-ovo** [comment]
Hey @karlhorky,  thanks for the reproduction steps, now the reuse of the existing patches file only takes effect when the version is specified, this problem will be solved after [this PR](https://github.com/pnpm/pnpm/pull/6197) merged.

**Comment by karlhorky** [comment]
I can confirm that this is working in the [v7.29.3 release](https://github.com/pnpm/pnpm/releases/tag/v7.29.3), thanks!

**Pull request merged**